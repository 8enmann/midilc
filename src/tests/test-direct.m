/*  Test direct selection
    Transposes a chord from one key to another through direct selection 
    @author Fred
    */

main(){
  Note a;
  Chord c;
  Sequence s;
  Number i;
  
  s = new_sequence();
  
  c = new_chord(C,E,G);
  s = s + transpose(c, C, C) + transpose(c, C, E) + transpose(c, C, C) + transpose(c, C, G);
  set_tempo(30);
  play(s);
}

transpose(input, first, second)
{
    Number n;
	n = first.pitch - second.pitch;
	return shift(input, n);
}

shift(c, n)
{
	Number i;
	Number len;
	len = c.length;
	for(i = 0; i < len; i=i+1)
	{
		c[i] = c[i] .+ n;
	}
	return c;
}
