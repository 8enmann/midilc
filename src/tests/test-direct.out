Tempo,30
0,4,60
0,4,64
0,4,67
4,4,56
4,4,60
4,4,63
8,4,60
8,4,64
8,4,67
12,4,53
12,4,57
12,4,60

